solr/solrj/src/java/org/apache/solr/client/solrj/request/JavaBinUpdateRequestCodec.java

```diff
@@ -128,7 +127,7 @@ public UpdateRequest unmarshal(InputStream is, final StreamingUpdateHandler hand
 
     // NOTE: if the update request contains only delete commands the params
     // must be loaded now
-    if (updateRequest.getParams() == null) {
+    if (updateRequest.getParams().iterator().hasNext() == false) { // no params
```

Comment:
This check is evil; took a while to find it as I debugged a test for hours.  Behavior is changing based on the presence of params or not.  Like nowhere is there a similar check (at least based on my adventure with this PR).  The history behind these lines of code is what I consider to be an ugly/messy solution to propagating a commitWithin param.  Tempting to add that on my long TODO list of messes to fix but our longer future is not "javabin", and the mess thankfully seems localized here.